File: clojure/subject-is-dead.clj
(ns subject-is-dead
  (:require [clojure.contrib.string :as str]))

(def subject "[Subject] is dead, long live [Subject]")
(def regex #"\[Subject\]")

(while (re-find regex subject)
       (def subject (str/replace-re regex subject subject))
       (print subject))

File: cpp/subject-is-dead.cpp
#include <iostream>
#include <regex>
#include <string>

using namespace std;

int main() {
    string subject = "[Subject] is dead, long live [Subject]";
    regex regex("\\[Subject\\]");

    while (regex_search(subject.begin(), subject.end(), regex)) {
        subject = regex_replace(subject, regex, subject);
        cout << subject;
    }
    return 0;
}
